paasta_tools/cli/cmds/status.py

```diff
@@ -1236,13 +1254,153 @@ def get_version_table_entry(
+        if unhealthy_replicas:
+            entry.append(f"    Unhealthy Replicas:")
+            replica_table = create_replica_table(unhealthy_replicas)
+            for line in replica_table:
+                entry.append(f"      {line}")
+    else:
+        replica_table = create_replica_table(replica_states)
+        for line in replica_table:
+            entry.append(f"    {line}")
     return entry
 
 
+class ReplicaState(Enum):
+    RUNNING = "Healthy"
+    UNSCHEDULED = "Unscheduled"
+    STARTING = "Starting"
+    WARMING_UP = "Warming Up"
+    TERMINATING = "Terminating"
+    HEALTHCHECK_FAILING = "Healthcheck Failing"
+    UNREACHABLE = "Unreachable"
+    UNHEALTHY = "Unhealthy"
+    UNKNOWN = "Unknown"
+
+    def is_unhealthy(self):
+        return self in ReplicaState.unhealthy_states()
+
+    @property
+    def color(self):
+        if self.is_unhealthy():
+            return PaastaColors.red
+        elif self.name == "RUNNING":
+            return PaastaColors.green
+        return PaastaColors.yellow
+
+    @classmethod
+    def unhealthy_states(cls):
+        return [cls.UNHEALTHY, cls.UNREACHABLE, cls.HEALTHCHECK_FAILING]
+
+
 # TODO(PAASTA-17287): Implement replica states
-def get_replica_states(pods: List[KubernetesPodV2]) -> str:
-    return PaastaColors.green(f"{len(pods)} Running")
+def get_replica_state(pod: KubernetesPodV2) -> ReplicaState:
+    phase = pod.phase
+    state = ReplicaState.UNKNOWN
+    if phase is None or not pod.scheduled:
+        state = ReplicaState.UNSCHEDULED
+        # If we are unscheduled but getting error unschedulable, is that unhealthy or still just unscheduled?
+    elif phase == "Pending":
+        # Check containers, liveness, restart
+        # if pod within healthcheck grace period, WARMING_UP
+        # if no containers scheduled yet, STARTING
+        if not pod.containers:
+            state = ReplicaState.STARTING
+        else:
+            state = ReplicaState.WARMING_UP
+    elif phase == "Running":
+        # Check pod restarts, current container statuses
+        # ReplicaState.HEALTHCHECK_FAILING?
+        # Check readiness probe
+        if not pod.ready:
+            state = ReplicaState.UNREACHABLE
+        else:
+            state = ReplicaState.RUNNING
+    elif phase == "Terminating":
+        state = ReplicaState.TERMINATING
+    elif phase == "Failed":
+        # e.g. pod.reason == evicted
+        state = ReplicaState.UNHEALTHY
+    return state
+
+
+def get_replica_states(
+    pods: List[KubernetesPodV2],
+) -> List[Tuple[ReplicaState, KubernetesPodV2]]:
+    return [(get_replica_state(pod), pod) for pod in pods]
+
+
+def create_replica_table(pods: List[Tuple[ReplicaState, Any]],) -> List[str]:
+    header = ["ID", "IP/Port", "Host deployed to", "Started at what localtime", "State"]
+    table: List[Union[List[str], str]] = [header]
+    for state, pod in pods:
+        # TODO: Is it always port 8888?
```

Comment:
should be based on the paasta contract.  I am not sure if we have this as a constant anywhere or anything, though

Reply:
TY that makes sense, deleted this comment.